CandySpaceTests/CandySpaceTests.swift

```diff
@@ -9,28 +9,22 @@ import XCTest
 @testable import CandySpace
 
 final class CandySpaceTests: XCTestCase {
-    // swiftlint:disable all
-    override func setUpWithError() throws {
-        // Put setup code here. This method is called before the invocation of each test method in the class.
-    }
+    var viewModel: GalleryViewModel?
+     override func setUp() {
+         super.setUp()
+         let networkManager = NetworkManager()
+         viewModel = GalleryViewModel(networkManager: networkManager)
+     }
 
-    override func tearDownWithError() throws {
-        // Put teardown code here. This method is called after the invocation of each test method in the class.
-    }
-
-    func testExample() throws {
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct results.
-        // Any test you write for XCTest can be annotated as throws and async.
-        // Mark your test throws to produce an unexpected failure when your test encounters an uncaught error.
-        // Mark your test async to allow awaiting for asynchronous code to complete. Check the results with assertions afterwards.
-    }
-
-    func testPerformanceExample() throws {
-        // This is an example of a performance test case.
-        self.measure {
-            // Put the code you want to measure the time of here.
-        }
-    }
-    // swiftlint:enable all
+     override func tearDown() {
+         super.tearDown()
+         viewModel = nil
+     }
+     func testGetImageGallery() async {
+         let expectedImages = 20
+         await viewModel!.getImageGallery(searchTerm: "nature")
+         XCTAssertEqual(viewModel?.images.count, expectedImages, "Hits are not equal")
+         await viewModel!.getImageGallery(searchTerm: "nature")
```

Comment:
same as line 25?

Reply:
This checks the cache. Well it is supposed to